packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/ParameterTransformations.java

```diff
@@ -17,7 +18,7 @@ public final class ParameterTransformations {
     private final List<ParameterTransformation> transformations;
 
     public ParameterTransformations(List<ParameterTransformation> parameterTransformations) {
-        this.transformations = parameterTransformations;
+        this.transformations = CollectionUtil.toImmutableList(parameterTransformations);
```

Comment:
The `CollectionUtil.toImmutableList` method is moved to here for clearer understanding of the correctness of equality checks. Since the implementation of equality requires that the list remains unchanged while its being executed, making this list immutable inside the class (instead of at call site) improves code clarity.